internal/lint/cronjob.go

```diff
@@ -31,6 +32,12 @@ func NewCronJob(co *issues.Collector, db *db.DB) *CronJob {
 	}
 }
 
+func CjSpecFor(fqn string, cj *batchv1.CronJob) rules.Spec {
```

Comment:
I think this could be dry-ed up a bit i.e coSpecFor(string, metav1.ObjectMetaAccessor, v1.PodSpec).
Thus we would not need the same impl for any pod resources.